plugins/destination/s3/client/write.go

```diff
@@ -49,6 +49,11 @@ func (c *Client) createObject(ctx context.Context, table *schema.Table, objKey s
 }
 
 func (c *Client) WriteTable(ctx context.Context, msgs <-chan *message.WriteInsert) error {
+	c.tableConcurrencyCh <- struct{}{}
```

Comment:
Depending on what you mean exactly this is a good objection and the PR should be closed.

I followed the `StreamBatchWriter` code and the channels where the writes are passed through are all unbuffered, so I'm pretty sure this line is effectively applying back-pressure on the process. But even if my code-reading skill is lacking, I see the back-pressure all the way to the source plugin (aws) output.

The heap usage on S3 went down by 42% and the others stayed the same, so it does seem like there is effectively less memory being used:

S3 with this PR
![S3 with concurrency experiment](https://github.com/cloudquery/cloudquery/assets/1078546/8034e66c-29ae-4845-99fb-205229a0bf23)

AWS with this PR
![AWS with concurrency experiment](https://github.com/cloudquery/cloudquery/assets/1078546/f60fff0e-d874-4c23-ad8e-d7ceb13ea078)

CLI with this PR
![CLI with concurrency experiment](https://github.com/cloudquery/cloudquery/assets/1078546/ddf2370c-7595-4401-b2bf-68c38624102b)

Would you mind peer-reviewing my experiment? It would be good to know if we should not merge this.